blockchain/v2/processor.go

```diff
@@ -124,6 +105,10 @@ func (state *pcState) enqueue(peerID p2p.ID, block *types.Block, height int64) e
 	return nil
 }
 
+func (state *pcState) height() int64 {
+	return state.context.tdState().LastBlockHeight
+}
+
 // purgePeer moves all unprocessed blocks from the queue
 func (state *pcState) purgePeer(peerID p2p.ID) {
 	// what if height is less than state.height?
```

Comment:
We might want to log (for debugging purpose) what blocks are removed from the queue once peer is purged.